modules/renter/hostdb/scan.go

```diff
@@ -136,10 +136,16 @@ func (hdb *HostDB) threadedProbeHosts() {
 	for hostEntry := range hdb.scanPool {
 		// Request settings from the queued host entry.
 		// TODO: use dialer.Cancel to shutdown quickly
-		hdb.log.Debugln("Scanning", hostEntry.NetAddress, hostEntry.PublicKey)
+		//
+		// A readlock is necessary when viewing the elements of the host entry.
+		hdb.mu.RLock()
+		netAddr := hostEntry.NetAddress
+		pubKey := hostEntry.PublicKey
+		hdb.mu.RUnlock()
```

Comment:
scanPool needs a *hostEntry so that it can modify the fields lol. We could add more safety with a getter and a setter, but I don't want to do too much of that without remodeling the entire concurrency scheme of the hostdb.

The hostdb is one of our earliest forays into high-concurrency (having multiple threads running all the time that all modify the state). There are a lot of mistakes and warts that are evidence of our inexperience.